integration-test/src/main/java/com/sequenceiq/it/cloudbreak/assertion/util/CloudProviderSideTagAssertion.java

```diff
@@ -105,7 +105,7 @@ public CloudProviderSideTagAssertion(CloudProviderProxy cloudProviderProxy, Tags
     }
 
     private void verifyTags(List<String> instanceIds, Map<String, String> customTags, TestContext testContext, String resourceName) {
-        if ((instanceIds != null && !instanceIds.isEmpty()) || !instanceIds.contains(null)) {
+        if ((instanceIds != null && !instanceIds.isEmpty()) && !instanceIds.contains(null)) {
```

Comment:
I think filtering out the nulls and verifying the non-nulls would be a better solution, maybe with a warning that there were null instance ids.